### DISCUSSION TITLE
What is the reason to uniquify the module names?

### DISCUSSION THREAD

**Discussion by davidmlw** [ucb-bar/chipyard#1532] [General]
I saw this python script trying to uniquify the module names. What is the reason behind? I suppose it is not an issue for any EDA flow. And EDA tools can do the job. If there are some reasons, could it be done in firtool?

https://github.com/ucb-bar/chipyard/blob/ca9e132bc61950f4d5e06b191316097a1642d641/scripts/uniquify-module-names.py#L185

**Comment by harrisonliew** [comment]
There was a discussion in #1388 and solved in PRs #1442 and #1452. TL;DR: because Chisel generates both the DUT and testbench, there would be module collisions between synthesized and testbench behavioral versions of the same module without uniquification.

We have an issue open in firtool: https://github.com/llvm/circt/issues/5066.

**Reply by davidmlw** [reply]
Many thanks. The answer is very clear! @harrisonliew
